src/pages/index.js

```diff
@@ -2,86 +2,168 @@
 
-const addImagePopup = new PopupWithForm(
+const avatarEditPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  (inputFieldValue, evt) => {
+    evt.preventDefault();
+    api
+      .setUserAvatar(inputFieldValue.Link)
+      .then((res) => {
+        userInfo.setUserInfo(res);
+        avatarEditPopup.closeAfterSubmit();
+      })
+      .catch((err) => alert(err))
+      .finally(() => avatarEditPopup.resetButtonText());
+  },
+  avatarEditFormValidator
+);
+avatarEditPopup.setEventListeners();
+
+const addCardPopup = new PopupWithForm(
   "#add-modal",
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    const { Link: link, Title: title } = inputFieldValues;
-    createCard({ name: title, link: link });
+    api
+      .uploadCard(inputFieldValues)
+      .then((res) => {
+        cardsListSection.addItem(
+          createCard({ name: inputFieldValues.Title, link: inputFieldValues.Link, _id: res._id }),
+          false
+        );
+        addCardPopup.closeAfterSubmit();
+      })
+      .catch((err) => alert(err))
+      .finally(() => addCardPopup.resetButtonText());
   },
   addFormValidator
 );
-addImagePopup.setEventListeners();
+addCardPopup.setEventListeners();
+
+api
+  .getInitialCards()
+  .then((res) => cardsListSection.renderItems(res, true))
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => userInfo.setUserInfo(res))
+  .catch((err) => alert(err));
 
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardsList = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      createCard(item);
-    },
-  },
-  cardsListSection
-);
-cardsList.renderItems();
+const confirmPopup = new PopupConfirm("#confirm-modal", (cardData, evt) => {
+  evt.preventDefault();
+  api
+    .deleteCard(cardData.cardId)
+    .then(() => {
+      cardData.cardEl.remove();
+      cardData.cardEl = null;
+      confirmPopup.close();
+    })
+    .catch((err) => alert(err))
+    .finally(() => confirmPopup.resetButtonText());
+});
+confirmPopup.setEventListeners();
 
 //add event listeners
-profileEditButton.addEventListener("click", openProfileForm);
-profileAddButton.addEventListener("click", openAddForm);
-
-//event listener callbacks
-function openProfileForm() {
+profileEditButton.addEventListener("click", () => {
   profilePopup.open();
-  const { name, title } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   profileModalNameInput.value = name;
-  profileModalDescInput.value = title;
+  profileModalDescInput.value = about;
   profileFormValidator.resetValidation();
   profileFormValidator.toggleButtonState();
-}
-
-function openAddForm() {
-  addImagePopup.open();
-}
+});
+profileAddButton.addEventListener("click", () => addCardPopup.open());
+profileAvatarEdit.addEventListener("click", () => {
+  avatarEditPopup.open();
+  avatarEditFormValidator.toggleButtonState();
+  avatarEditFormValidator.resetValidation();
```

Comment:
`avatarEditFormValidator.resetValidation();` is not needed when you open the avatar form because you don't need to check validity and show errors under the input on opening 


Actually, you should add `this.toggleButtonState();` inside `FormValidator` inside method `enableValidation` to initially disable the button not to duplicate it on opening 


```js
  enableValidation() {
    this._setEventListeners();
    this.toggleButtonState();
  }

```


So, you don't need anything else here in `index.js` 


```js
profileAvatarEdit.addEventListener("click", () => {
  avatarEditPopup.open();
});

```